yt/frontends/amrvac/tests/test_outputs.py

```diff
@@ -21,118 +26,113 @@
-    # dataset with dust fields
-    ds = data_dir_load(rmi_cartesian_dust_2D)
-    for test in small_patch_amr(ds, _get_fields_to_check(ds)):
-        test_rmi_cartesian_dust_2D.__name__ = test.description
-        yield test
+    # return [fields, field_ids]
+    return [ds, fields]
+
+
+def get_pairs():
+    f_list = [_get_fields_to_check(fname) for fname in ds_list]
+    pairs = [(i[0], f) for i in f_list for f in i[1]]
+    return pairs
+
+
+@pytest.mark.answer_test
+class TestAMRVAC:
+    answer_file = None
+    saved_hashes = None
+
+    @requires_file(khi_cartesian_2D)
+    def test_AMRVACDataset(self):
+        assert isinstance(utils.data_dir_load(khi_cartesian_2D), AMRVACDataset)
+
+    @utils.requires_ds(blastwave_cartesian_3D)
+    def test_domain_size(self):
+        # "Check for correct box size, see bw_3d.par"
+        ds = utils.data_dir_load(blastwave_cartesian_3D)
+        for lb in ds.domain_left_edge:
+            assert int(lb) == 0
+        for rb in ds.domain_right_edge:
+            assert int(rb) == 2
+        for w in ds.domain_width:
+            assert int(w) == 2
+
+    @requires_file(blastwave_cartesian_3D)
+    def test_grid_attributes(self):
+        # "Check various grid attributes"
+        ds = utils.data_dir_load(blastwave_cartesian_3D)
+        grids = ds.index.grids
+        assert ds.index.max_level == 2
+        for g in grids:
+            assert isinstance(g, AMRVACGrid)
+            assert isinstance(g.LeftEdge, YTArray)
+            assert isinstance(g.RightEdge, YTArray)
+            assert isinstance(g.ActiveDimensions, np.ndarray)
+            assert isinstance(g.Level, (np.int32, np.int64, int))
+
+    @pytest.mark.usefixtures("hashing")
+    @pytest.mark.parametrize("ds", ds_list, indirect=True)
+    def test_gh_pr(self, ds):
+        self.hashes.update({"grid_hierarchy": grid_hierarchy(ds)})
+        self.hashes.update({"parentage_relationships": parentage_relationships(ds)})
+
+    @pytest.mark.usefixtures("hashing")
+    @pytest.mark.parametrize("ds, f", get_pairs(), indirect=True)
+    def test_gv(self, f, ds):
+        if ds and f is None:
+            pytest.skip("Data not found.")
+        self.hashes.update({"grid_values": grid_values(ds, f)})
+
+    @pytest.mark.usefixtures("hashing")
+    @pytest.mark.parametrize("ds, f", get_pairs(), indirect=True)
+    @pytest.mark.parametrize("d", dso_list, indirect=True)
+    def test_fv(self, d, f, ds):
+        if ds and f is None:
+            pytest.skip("Data not found.")
+        self.hashes.update({"field_values": field_values(ds, f, d)})
+
+    @pytest.mark.usefixtures("hashing")
+    @pytest.mark.parametrize("ds, f", get_pairs(), indirect=True)
+    @pytest.mark.parametrize("d", dso_list, indirect=True)
+    @pytest.mark.parametrize("a", a_list, indirect=True)
+    @pytest.mark.parametrize("w", w_list, indirect=True)
+    def test_pv(self, a, d, w, f, ds):
+        if ds and f is None:
+            pytest.skip("Data not found.")
```

Comment:
Once again I'm having a hard time understanding what's happening here (because I can't read get_pairs). I would suggest specifying in the skip message exactly what data is missing but I'm not sure how.